test/browser_tests/cucumber/step_definitions/reusable-text-snippets.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+
+var util = require( '../../util/index.js' );
+var shouldShouldnt = util.shouldShouldnt;
+
+var { defineSupportCode } = require( 'cucumber' );
+var { expect } = require( 'chai' );
+
+
+defineSupportCode( function( { When, Then } ) {
+
+  When( 'there exists a reusable text snippet on this page',
+    function() {
+      var rts = element( by.css( '.m-reusable-text-snippet' ) );
+      return expect( rts.isPresent() ).to.eventually.equal( true );
+    }
+  );
+
+  Then( /the snippet output (should|shouldn't) include a slug-style header/,
+    function( includeHeader ) {
+      var rts;
+
+      if ( shouldShouldnt( includeHeader ) ) {
+        rts = element.all( by.css( '.m-reusable-text-snippet' ) ).first();
```

Comment:
Thanks for the comment! I myself wasn't wild about having to choose them with `.first()` and `.last()`. You're right that searching on content would make it easier to know what's going on. We decided to go this route because searching by content isn't something that Protractor natively supports. We'd have to do something like [this custom function around `by.xPath`](https://stackoverflow.com/a/36334925/279566), and it seemed like it wasn't worth it. What do you think?
2.